cpp/src/arrow/memory_pool_internal.h

```diff
@@ -26,7 +26,7 @@ namespace memory_pool {
 
 namespace internal {
 
-static constexpr size_t kAlignment = 64;
+static constexpr size_t kAlignment = 512;
```

Comment:
This is not going to work.  The impact would be too large.  Is this because you are using direct I/O below?  If so,we could potentially make the alignment configurable and create a custom memory pool and require users that want direct I/O to use the custom memory pool.

Reply:
Yeah I was going to add the option later (I wanted to change it to this for now just so I could test other stuff).